--- keystone.orig	2018-01-09 10:50:01.005819940 +0000
+++ keystone	2018-01-09 10:50:18.133820750 +0000
@@ -607,7 +607,7 @@
         --bootstrap-role-name admin \
         --bootstrap-service-name keystone \
         --bootstrap-region-id "$REGION_NAME" \
-        --bootstrap-admin-url "$KEYSTONE_AUTH_URI" \
+        --bootstrap-admin-url "$KEYSTONE_AUTH_URI/v2.0" \
         --bootstrap-public-url "$KEYSTONE_SERVICE_URI"
 }
