--- /opt/stack/contrail/devstack/plugin.sh.orig	2018-01-12 11:52:36.534751974 +0000
+++ /opt/stack/contrail/devstack/plugin.sh	2018-01-12 12:15:53.605687810 +0000
@@ -166,21 +166,21 @@
     [ ! -d /var/log/contrail ] && sudo mkdir /var/log/contrail
     sudo chmod 777 /var/log/contrail

-    run_process vrouter "sudo contrail-vrouter-agent --config_file=/etc/contrail/contrail-vrouter-agent.conf"
-    run_process api-srv "contrail-api --conf_file /etc/contrail/contrail-api.conf"
+    run_process vrouter "/usr/bin/contrail-vrouter-agent --config_file=/etc/contrail/contrail-vrouter-agent.conf" ubuntu root
+    run_process api-srv "/usr/bin/contrail-api --conf_file /etc/contrail/contrail-api.conf"
     # Wait for api to be ready, as it creates cassandra CF required for disco to start
     is_service_enabled disco && is_service_enabled api-srv && wget --no-proxy --retry-connrefused --no-check-certificate --waitretry=1 -t 60 -q -O /dev/null http://$APISERVER_IP:8082 || true
-    run_process disco "contrail-discovery --conf_file /etc/contrail/contrail-discovery.conf"
-    run_process svc-mon "contrail-svc-monitor --conf_file /etc/contrail/contrail-svc-monitor.conf"
-    run_process schema "contrail-schema --conf_file /etc/contrail/contrail-schema.conf"
-    run_process control "sudo contrail-control --conf_file /etc/contrail/contrail-control.conf"
-    run_process collector "contrail-collector --conf_file /etc/contrail/contrail-collector.conf"
-    run_process analytic-api "contrail-analytics-api --conf_file /etc/contrail/contrail-analytics-api.conf"
+    #run_process disco "contrail-discovery --conf_file /etc/contrail/contrail-discovery.conf"
+    run_process svc-mon "/usr/bin/contrail-svc-monitor --conf_file /etc/contrail/contrail-svc-monitor.conf"
+    run_process schema "/usr/bin/contrail-schema --conf_file /etc/contrail/contrail-schema.conf"
+    run_process control "/usr/bin/contrail-control --conf_file /etc/contrail/contrail-control.conf" ubuntu root
+    run_process collector "/usr/bin/contrail-collector --conf_file /etc/contrail/contrail-collector.conf"
+    run_process analytic-api "/usr/bin/contrail-analytics-api --conf_file /etc/contrail/contrail-analytics-api.conf"
     run_process alarm-gen "/usr/bin/contrail-alarm-gen --conf_file /etc/contrail/contrail-alarm-gen.conf"
-    run_process query-engine "contrail-query-engine --conf_file /etc/contrail/contrail-query-engine.conf"
-    run_process dns "contrail-dns --conf_file /etc/contrail/dns/contrail-dns.conf"
+    run_process query-engine "/usr/bin/contrail-query-engine --conf_file /etc/contrail/contrail-query-engine.conf"
+    run_process dns "/usr/bin/contrail-dns --conf_file /etc/contrail/dns/contrail-dns.conf"
     #NOTE: contrail-dns checks for '/usr/bin/contrail-named' in /proc/[pid]/cmdline to retrieve bind status
-    run_process named "sudo /usr/bin/contrail-named -g -c /etc/contrail/dns/contrail-named.conf"
+    run_process named "/usr/bin/contrail-named -g -c /etc/contrail/dns/contrail-named.conf -u ubuntu" ubuntu root

     run_process ui-jobs "cd $CONTRAIL_DEST/contrail-web-core; sudo nodejs jobServerStart.js"
     run_process ui-webs "cd $CONTRAIL_DEST/contrail-web-core; sudo nodejs webServerStart.js"
