--- /opt/stack/contrail/devstack/lib/contrail_config	2017-07-27 09:12:11.603603000 +0000
+++ /home/ubuntu/contrail_plugin_zioc/contrail_config.new	2017-07-28 07:25:46.151703981 +0000
@@ -340,6 +340,35 @@
     iniset -sudo $config_file DEFAULT log_local 1
 }
 
+function contrail_config_alarm_gen()
+{
+    local config_file="/etc/contrail/contrail-alarm-gen.conf"
+    sudo truncate -s 0 $config_file
+
+    [[ -n $CLUSTER_ID ]] && iniset -sudo $config_file DATABASE cluster_id $CLUSTER_ID
+
+    iniset -sudo $config_file DEFAULT host_ip $COLLECTOR_IP
+
+    #REVISIT: analytics api currently fails with default value (15)
+    iniset -sudo $config_file DEFAULT partitions 0
+
+    if _vercmp $CONTRAIL_BRANCH "<" R3.0; then
+        iniset -sudo $config_file DEFAULT cassandra_server_list "$CASSANDRA_IP_PORT_LIST"
+    else
+        iniset -sudo $config_file DEFAULT cassandra_server_list "$CASSANDRA_CQL_IP_PORT_LIST"
+    fi
+
+    if _vercmp $CONTRAIL_BRANCH "<" R4.0; then
+        iniset -sudo $config_file DISCOVERY disc_server_ip $DISCOVERY_IP
+        iniset -sudo $config_file DISCOVERY disc_server_port 5998
+    else
+        iniset -sudo $config_file DEFAULT collectors "$COLLECTOR_IP_PORT_LIST"
+    fi
+
+    iniset -sudo $config_file DEFAULT log_level 'SYS_DEBUG'
+    iniset -sudo $config_file DEFAULT log_local 1
+}
+
 function contrail_config_query()
 {
     local config_file="/etc/contrail/contrail-query-engine.conf"
